src/mame/drivers/dfs500.cpp

```diff
@@ -0,0 +1,1036 @@
+	map(0x4000, 0x4001).mirror(0x8ff0).r(FUNC(dfs500_state::RB0_r));                                                // "RB0" IC32/IC33
+	map(0x4002, 0x4003).mirror(0x8ff0).r(FUNC(dfs500_state::RB1_r));                                                // "RB1" IC55
+	map(0x4004, 0x4005).mirror(0x8ff0).r(FUNC(dfs500_state::RB2_r));                                                // "RB2" IC56
+	map(0x5000, 0x5001).mirror(0x8ff0).w(FUNC(dfs500_state::WB0_w));                                                // "WB0" IC26/IC27
+}
+
+void dfs500_state::effectcpu_mem_map(address_map &map)
+{
+	// Note: As far as I can tell by the schematics in the service manual, the ram mirror should be 0x90000.
+	//       But I see unmapped read acesses to 0x4800, which induces me to make the mirror 0x94000, instead.
+	//	 FIXME: This should be double-checked!
+	map(0x00000, 0x03fff).mirror(0x94000).ram();                        // 2x 8kb SRAM chips at IC23/IC24 
+	map(0x08000, 0x0bfff).mirror(0x90000).bankr("rombank1");            // Effect data on 4x 512kb EPROMs at IC5/IC6/IC7/IC8
+	map(0x0c000, 0x0ffff).mirror(0x90000).bankr("rombank2");            // Second banked view of the contents of the same effect data EPROMs
+	map(0x20000, 0x20001).mirror(0x9fffe).noprw(); // FIXME: Do something with the MTRX signal generated by this memory access
+	map(0x46000, 0x46fff).mirror(0x91000).ram().share("sgram"); // IC80/IC81 at FM-29 (4/6)
+	                                                            // Selected by IC201 at FM-29 (3/6)
+	//map(0x68000, 0x68001).mirror(0x907fe).rw(FUNC(dfs500_state::RA0_r), FUNC(dfs500_state::WA0_w)); // "RA0" IC26/IC27 & "WA0" IC32/IC33
+	                                                                                                // 16-bit data latches for communication between CPUs
+	//map(0x68800, 0x68801).mirror(0x907fe).r(FUNC(dfs500_state::RA1_r));            // "RA1" IC25/IC64
+	//map(0x69000, 0x69000).mirror(0x907ff).w(FUNC(dfs500_state::rombank2_entry_w)); // "WA2" IC29
+	//map(0x69800, 0x69800).mirror(0x907ff).w(FUNC(dfs500_state::rombank1_entry_w)); // "WA3" IC30
+	//map(0x6A000, 0x6A000).mirror(0x907ff).w(FUNC(dfs500_state::input_select_w));   // "WA4" IC31
+	map(0x70000, 0x7ffff).mirror(0x80000).rom().region("effectcpu", 0); // 2x 64kb EPROMs at IC3/IC4,
+	                                                                    // Note: the 1st half of each is entirely made of 0xFF
+	
+	// ==== "ORG1" registers ====
+	// "controlsignals" (Not sure yet of their actual use)
+	// D13: PS2
+	// D12: PS1
+	// D11: FGS1
+	// D10: FGS0
+	// D9: BGS1
+	// D8: BGS0
+	// D7: TKON
+	// D6: TKCONT
+	// D5: AM2
+	// D4: AM1
+	// D3: AM0
+	// D2: BM2
+	// D1: BM1
+	// D0: BM0
+	map(0x6B078, 0x6B079).mirror(0x90700).w("controlsignals", FUNC(generic_latch_16_device::write));
+
+	// "Reg7_5":
+	//  Not sure yet of their actual use...
+	map(0x6B07a, 0x6B07b).mirror(0x90700).w("reg7_5", FUNC(generic_latch_16_device::write));
```

Comment:
Please be consistent in your capitalization of hexadecimal constants. Here you have mixed both upper-case and lower-case in the same constant, even. Convention is typically to use lower-case hexadecimal constants.